component/loki/source/cloudflare/internal/cloudflaretarget/fields.go

```diff
@@ -39,10 +40,38 @@ var (
 		"FirewallMatchesActions", "FirewallMatchesRuleIDs", "OriginResponseBytes", "OriginResponseTime", "ClientDeviceType", "WAFFlags", "WAFMatchedVar", "EdgeColoID",
 		"RequestHeaders", "ResponseHeaders",
 	}...)
+	allFieldsMap = buildAllFieldsMap(allFields)
 )
 
-// Fields returns the mapping of FieldsType to the set of fields it represents.
-func Fields(t FieldsType) ([]string, error) {
+// Fields returns the concatenation of set of fields represented by the Fieldtype and the give custom fields without duplicates.
+func Fields(t FieldsType, customFields []string) ([]string, error) {
+	fieldsSubset, err := getFieldSubset(t)
+	if err != nil {
+		return nil, err
+	}
+	return mergeAndRemoveDuplicatedFields(fieldsSubset, customFields), nil
+}
+
+func mergeAndRemoveDuplicatedFields(fieldsSubset, customFields []string) []string {
```

Comment:
Nit - a bit shorter name
```suggestion
func mergeAndRemoveDuplicates(fieldsSubset, customFields []string) []string {
```